apps/personal/contacts/main.py

```diff
@@ -4,57 +4,113 @@
+        for c in all_contacts:
+            menu_entries.append([c.short_name(), lambda x=c:
+                                 self.open_contact_details_page(x)])
+
+        return menu_entries
 
-    def create_contact_page(self, contact):
+    def open_contact_details_page(self, contact):
         # type: (Contact) -> None
-        contact_attrs = [getattr(contact, a) for a in contact.get_filled_attributes()]
+        contact_attrs = [getattr(contact, a) for a in
+                         contact.get_filled_attributes()]
         Listbox(i=self.i, o=self.o, contents=contact_attrs).activate()
 
-
-def find_contact_files(folder):
-    # type: (str) -> list(str)
-    home = os.path.expanduser(folder)
-    if not os.path.exists(home):
-        os.mkdir(home)
-    contact_card_files = [os.path.join(home, f) for f in os.listdir(home) if f.lower().endswith("vcf")]
-    return contact_card_files
-
-
-def load_vcf(folder):
-    # type: (str) -> None
-    contact_card_files = find_contact_files(folder)
-    contacts = VCardContactConverter.from_vcards(contact_card_files)
-
-    address_book = AddressBook()
-    for contact in contacts:
-        address_book.add_contact(contact)
-    address_book.save_to_file()
-    logger.info("Saved to {}".format(address_book.get_save_file_path()))
-
+    def open_actions_menu(self):
+        menu_contents = [
+            ["Settings", lambda: self.open_settings_page()],
+            ["CardDAV Import", lambda: self.synchronize_carddav()],
+            ["Reset address book", lambda: self.reset_addressbook()]
+        ]
+        Menu(menu_contents, i=self.i, o=self.o, name="My menu").activate()
+
+    def open_settings_page(self):
+        if self.vdirsyncer_executable:
+            vdirsyncer_executable = self.vdirsyncer_executable
+        else:
+            vdirsyncer_executable = 'Not found'
+
+        attrs = [
+            ["-- VdirSyncer"],
+            ["Executable: {}".format(vdirsyncer_executable) ],
+            ["Config: {}".format(VDIRSYNCER_CONFIG)]
+        ]
+        Listbox(i=self.i, o=self.o, contents=attrs).activate()
+
+    def reset_addressbook(self):
+        alert = "This action will delete all of your contacts. Are you sure?"
+        do_reset = DialogBox('yc', i=self.i, o=self.o, message=alert,
+                             name="Address book reset").activate()
+
+        if do_reset:
+            self.address_book.reset()
+            announce = "All of your contacts were deleted."
+            Printer(announce, i=self.i, o=self.o, sleep_time=2, skippable=True)
+            # Reload the now empty address book
+            self.reload()
+
+    def synchronize_carddav(self):
+        if (not os.path.isfile(self.vdirsyncer_executable) or
+        not os.access(self.vdirsyncer_executable, os.X_OK)):
+            Printer('Could not execute vdirsyncer.', i=self.i, o=self.o,
+                    sleep_time=2, skippable=True)
+            callback()
```

Comment:
Wait, why is there a `callback()` call? As long as you're using `Menu`, you'll return back to it once the code has finished executing.

Reply:
```suggestion
            return
```

Reply:
Fixed by bb19598.